lib/phase/structural.js

```diff
@@ -49,22 +48,22 @@ function checkAdditionalRules(data) {
  * @param   {Object}   doc      Direct document from YAML parsing
  * @returns {Promise}
  */
-module.exports = doc => (
-    new Promise((resolve, reject) => {
-        Joi.validate(doc, SCHEMA_CONFIG, {
+async function structuralValidation(doc) {
+    try {
+        const data = await SCHEMA_CONFIG.validateAsync(doc, {
             abortEarly: false
-        }, (schemaErr, data) => {
-            if (schemaErr) {
-                return reject(schemaErr);
-            }
+        });
 
-            const errors = checkAdditionalRules(data);
+        const errors = checkAdditionalRules(data);
 
-            if (errors) {
-                return reject(errors);
-            }
+        if (errors) {
+            throw errors;
+        }
 
-            return resolve(data);
-        });
-    })
-);
+        return data;
+    } catch (err) {
```

Comment:
why is this catch needed ? can't the entire `try..catch` be omitted and let the function throw err